src/clj/harja/palvelin/palvelut/kulut/valikatselmukset.clj

```diff
@@ -362,25 +441,46 @@
         tavoitehinta (valikatselmus-q/hae-oikaistu-tavoitehinta db {:urakka-id urakka-id
                                                                     :hoitokauden-alkuvuosi hoitokauden-alkuvuosi})
         kattohinta (valikatselmus-q/hae-oikaistu-kattohinta db {:urakka-id urakka-id
-                                                                :hoitokauden-alkuvuosi hoitokauden-alkuvuosi})]
+                                                                :hoitokauden-alkuvuosi hoitokauden-alkuvuosi})
+        erilliskustannus_id (tallenna-lupausbonus db tiedot kayttaja)
+        sanktio_id (tallenna-lupaussanktio db tiedot kayttaja)]
     (case paatoksen-tyyppi
       ::valikatselmus/tavoitehinnan-ylitys (tarkista-tavoitehinnan-ylitys tiedot tavoitehinta kattohinta)
       ::valikatselmus/kattohinnan-ylitys (tarkista-kattohinnan-ylitys tiedot urakka)
       ::valikatselmus/tavoitehinnan-alitus (tarkista-tavoitehinnan-alitus db tiedot urakka tavoitehinta hoitokauden-alkuvuosi)
       ::valikatselmus/lupausbonus (tarkista-lupausbonus db kayttaja tiedot)
       ::valikatselmus/lupaussanktio (tarkista-lupaussanktio db kayttaja tiedot))
-    (valikatselmus-q/tee-paatos db (tee-paatoksen-tiedot tiedot kayttaja hoitokauden-alkuvuosi))))
+    (valikatselmus-q/tee-paatos db (tee-paatoksen-tiedot tiedot kayttaja hoitokauden-alkuvuosi erilliskustannus_id sanktio_id))))
 
-(defn poista-paatos
-  [db kayttaja {::valikatselmus/keys [paatoksen-id] :as tiedot}]
-  (oikeudet/vaadi-kirjoitusoikeus oikeudet/urakat-suunnittelu-kustannussuunnittelu
-    kayttaja
-    (::urakka/id tiedot))
+(defn- poista-liitetty-lupausbonus [db urakka-id bonus-id kayttaja-id]
+  (log/debug "poista-liitetty-lupausbonus :: tää ei tee vielä mitään : urakka-id bonus-id kayttaja-id : " urakka-id bonus-id kayttaja-id)
+  (erilliskustannus-kyselyt/poista-erilliskustannus! db {:id bonus-id :kayttaja-id kayttaja-id}))
+
+(defn- poista-liitetty-lupaussanktio [db urakka-id sanktio-id kayttaja-id]
+  (let [sanktio (first (sanktiot-q/hae-sanktio db sanktio-id))
+        laatupoikkeama (first (laatupoikkeamat-q/hae-laatupoikkeaman-tiedot db {:urakka urakka-id
+                                                                                :id (:laatupoikkeama-id sanktio)}))
+        _ (sanktiot-q/poista-sanktio! db {:id (:id sanktio) :muokkaaja kayttaja-id})
```

Comment:
Tässä kannattaisi hyödyntää uutta apufunktiota laadunseurannasta:
`harja.palvelin.palvelut.laadunseuranta/poista-suorasanktio`

Se ottaa huomioon myös maksuerän likaiseksi merkitsemisen ym.

Reply:
Erittäin hyvä. Kiitos!